modelon/impact/client/sal/workspace.py

```diff
@@ -226,24 +209,20 @@ def get_published_workspaces(
         type: Optional[str] = None,
         group_name: Optional[str] = None,
     ) -> Dict[str, Any]:
-        query = {}
-        if name:
-            query["workspaceName"] = name
-        if has_data:
-            query["hasData"] = str(has_data)
+        query = {
+            "workspaceName": name,
+            "hasData": has_data,
+            "ownerUsername": owner_username,
+            "type": type,
+            "groupName": group_name,
+        }
```

Comment:
Why did you skip all the conditions? It seems unnecessary to pass, e.g.,  `workspaceName=None` all the way to the server?

Reply:
I felt this makes the python client api consistent, if the defaults in server are updated for some reason? The server would anyway parse the query strings and set defaults if they are not set here.  I agree for workspaceName probably it doesnt make sense to send in.

Reply:
I agree with this argument for boolean fields (like  `has_data`) and eventually pagination control (first/max) but I feel it is a bit weird for optional strings like workspaceName, ownerUsername and groupName. Are those getting serialized like `workspaceName=null` when passed to requests?

Reply:
Just read the docs, seems like requests would not append it to the URL if its **NONE** its seems as per docs - https://docs.python-requests.org/en/latest/user/quickstart/#passing-parameters-in-urls

For empty string they would just be added like https://blahblah.com/api?workspaceName=&ownerUsername=

Reply:
Thanks for double-checking! Then I'm fine with the update.